### ISSUE TITLE
sounddevice.PortAudioError: Error opening OutputStream: Device unavailable [PaErrorCode -9985]

### ISSUE THREAD

**Issue by cool-RR** [spatialaudio/python-sounddevice#199]
I connected a USB headset to listen to a sound made by `sounddevice` as part of #198, and I got the following error:

```
Traceback (most recent call last):
  File "foo.py", line 45, in <module>
    play(sample_rate)
  File "foo.py", line 39, in play
    sounddevice.play(array_2, samplerate=hz)
  File "C:\Program Files\Python37\lib\site-packages\sounddevice.py", line 154, in play
    **kwargs)
  File "C:\Program Files\Python37\lib\site-packages\sounddevice.py", line 2417, in start_stream
    **kwargs)
  File "C:\Program Files\Python37\lib\site-packages\sounddevice.py", line 1374, in __init__
    **_remove_self(locals()))
  File "C:\Program Files\Python37\lib\site-packages\sounddevice.py", line 780, in __init__
    'Error opening {0}'.format(self.__class__.__name__))
  File "C:\Program Files\Python37\lib\site-packages\sounddevice.py", line 2572, in _check
    raise PortAudioError(errormsg, err)
sounddevice.PortAudioError: Error opening OutputStream: Device unavailable [PaErrorCode -9985]
```

The headset otherwise works, Windows 7 plays audio through it just fine.

**Comment by mgeier** [comment]
Probably the device only supports a certain combination of `samplerate` and `channels`?

You can check this with [sd.check_output_settings()](https://python-sounddevice.readthedocs.io/en/0.3.13/api.html#sounddevice.check_output_settings).

Or probably the device is blocked by Windows? Can you de-select it in the Windows audio settings?

**Comment by cool-RR** [comment]
Running `sd.check_output_settings` returns `None`.

I opened the Sound control panel item. I don't know what you meant by "deselect", but I right-clicked it and hit Test, and Windows said "The device is being used by another application.  Please close any devices that are playing audio to this device and then try again." I just connected the device and I'm not playing anything on it. To be safe I closed Skype, but I still got that message. Any idea why that's happening? 

Spoiler, I'm thinking of giving a talk using `sounddevice`, and I'll be connecting to the venue's sound system, and I dread for my talk to be ruined by this problem.

**Comment by mgeier** [comment]
> I don't know what you meant by "deselect",

Me neither, to be honest. I just wanted you to look into the Windows audio settings and try a few things.

I'm not a Windows user, so I have no clue what can be done in there.

> and Windows said "The device is being used by another application. Please close any devices that are playing audio to this device and then try again."

Well that's basically the same message as the error you mentioned above: "Device unavailable". So that seems to be the original problem.

Probably there is a crashed process that's still occupying the sound device?

Have you tried turning it off and on again (https://www.youtube.com/watch?v=p85xwZ_OLX0)?

> I'll be connecting to the venue's sound system

I guess you wouldn't use your USB headset for that?

You should probably test with the device that you'll be actually using.

> and I dread for my talk to be ruined by this problem.

Using audio when presenting on someone else's system is quite likely to fail anyway. You should make sure to have a backup plan.

**Comment by cool-RR** [comment]
Thanks for your help!

**Comment by cool-RR** [comment]
When troubleshooting another issue with @mgeier , we came to the conclusion that this bug only happens when ASIO is included in the PortAudio DLL, and is probably related to the initialization action that PortAudio does to ASIO.

**Comment by cool-RR** [comment]
Headset model: Logitech USB Headset H390.

Hardware IDs:

```
USB\VID_046D&PID_0A44&REV_0127&MI_00
USB\VID_046D&PID_0A44&MI_00
```
